mega-linter-runner/generators/mega-linter/templates/mega-linter.yml

```diff
@@ -41,16 +41,17 @@ jobs:
     runs-on: ubuntu-latest
 
     # Give the default GITHUB_TOKEN write permission to commit and push, comment
-    # issues, and post new Pull Requests; remove the ones you do not need
+    # issues & post new PR; remove the ones you do not need
     permissions:
       contents: write
       issues: write
       pull-requests: write
 
     steps:
+
       # Git Checkout
       - name: Checkout Code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
```

Comment:
No longer relevant after https://github.com/oxsecurity/megalinter/pull/3032/files#diff-b92ed68a80095754009f7d140c5f743d495098408c2fdc6873a6bf525a4ecb87R147-R155

Reply:
@echoix one thing to consider with an upgrade to `checkout@v4` is it _may_ break workflows for people who use local linting. I'm thinking about tools like https://github.com/nektos/act - they're falling pretty far behind in keeping up with GitHub on server images. @catthehacker is the user that was maintaining those and they have left GitHub.  It is possible that an upgrade to @v4, which, if I recall correctly upgrades the node version required, could break MegaLinter's ability to run locally via `act`.

It's something we need to earmark for a test (and I have with the change I proposed) prior to merging this into MegaLinter's next release.

Reply:
I think that the most important use case isn't with act, and thus, it way more important to not use a node version that is EOL since Sept 11, 2023 (node 16 used in actions/checkout@v3). Running locally with act, even if I have done it a couple of times in the past (when doing big changes in the docker building actions and multiplatform builds, iterating on GitHub was way too long), remains more of a hack than a supported (by us) use-case. Having an up to date runner image for act seems like the way to solve your interrogation.

I don't understand the meaning of your last sentence, especially "earmark for a test", and what do you expect before a next release (that should be due soon IMO, maybe wait a little after we settled and merged the big template changes and other link changes PRs).

Reply:
We should have pretty up-to-date nodejs on images.

Reply:
> I think that the most important use case isn't with act, and thus, it way more important to not use a node version that is EOL since Sept 11, 2023 (node 16 used in actions/checkout@v3). Running locally with act, even if I have done it a couple of times in the past (when doing big changes in the docker building actions and multiplatform builds, iterating on GitHub was way too long), remains more of a hack than a supported (by us) use-case. Having an up to date runner image for act seems like the way to solve your interrogation.
> 
> I don't understand the meaning of your last sentence, especially "earmark for a test", and what do you expect before a next release (that should be due soon IMO, maybe wait a little after we settled and merged the big template changes and other link changes PRs).

Fair point. I'm growing more and more concerned about the `act` upkeep, even though I think it's a great tool :( not enough community engagement. Don't let it slow MegaLinter down!